### PR TITLE
Automatically set the file extension to use with the git adapter

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

File: src/storage-adapters/git/index.js [hunk 1]
```diff
@@ -13,12 +13,13 @@ import Git from './git.js';
...
 export default class GitAdapter {
-  constructor({ path, fileExtension, author, publish, prefixMessageToSnapshotId }) {
+  constructor({ path, author, publish, prefixMessageToSnapshotId }) {
     this.path = path;
     this.author = author;
-    this.fileExtension = fileExtension;
     this.needsPublication = publish;
     this.prefixMessageToSnapshotId = prefixMessageToSnapshotId;
+
+    mime.define({ 'text/markdown': ['md'] }, true); // ensure extension for markdown files is `.md` and not `.markdown`
```
Comment (martinratinaud) [inline src/storage-adapters/git/index.js:22]: Does this has to be in the constructor?
Seems to me it's more global and should be below imports

File: src/storage-adapters/git/index.test.js [hunk 2]
```diff
@@ -45,7 +45,6 @@ describe('GitAdapter', () => {
     subject = new GitAdapter({
```
Comment (martinratinaud) [inline src/storage-adapters/git/index.test.js:45]: subject has been a bit confusing for me to understand.

What about `adapter`?

**Review by martinratinaud — APPROVED** [review]
LGTM

File: src/storage-adapters/git/index.test.js (hunk 2 above)
Comment (Ndpnt) [reply src/storage-adapters/git/index.test.js:45]: It's a common practice used through the whole codebase so I won't modify it here but we could discuss about this if we want to change this globally.

**Pull request merged**